protelis/protelis-interpreter/src/main/java/org/protelis/vm/impl/SimpleExecutionEnvironment.java

```diff
@@ -14,6 +17,7 @@
 public final class SimpleExecutionEnvironment implements ExecutionEnvironment {
 
     private final Map<FasterString, Object> env = new LinkedHashMap<>();
+    private final Set<String> keys = new HashSet<>();
```

Comment:
I was conservative in my change here, because I wasn't sure if there was a strong reason for keeping FasterString.  If you're comfortable switching it over, I'll do that and check in shortly.